### PR TITLE
v14.3.1 build fixes

### LINKED ISSUE #1: Support different NetCDF paths for C and Fortran
Note that not setting `GC_BIN`, `GC_F_BIN`, `GC_INCLUDE`, `GC_F_INCLUDE`, `GC_LIB`, and `GC_F_LIB` environment variables correctly leads to build failures. As I demonstrated in https://github.com/fetch4/geos-chem/pull/1, this can be reduced to two environment variables: the NetCDF home paths for C and Fortran.

Currently, the `modelErc` created by GISS only defines `NETCDFHOME`, which I believe corresponds to the Fortran libraries. It would be better if two paths could be set, to account for the case where the C and Fortran paths differ.

### LINKED ISSUE #2: Fix v14.3.1 CMake build
Currently, building GISS-GC on branch `dev/giss-gc-14` gets 92% of the way through the GEOS-Chem CMake build before crashing with
```
Error copying Fortran module "mod/get_met_mod.mod".  Tried "mod/GET_MET_MOD.mod" and "mod/get_met_mod.mod".
```

This will need fixing in order to progress with the version upgrade.

### PR DESCRIPTION
**Pull request by joewallwork**
Closes #2.
Closes #1.
Closes https://github.com/fetch4/geos-chem/issues/2.
Linked PRs:
https://github.com/fetch4/geos-chem/pull/3.
https://github.com/fetch4/HEMCO/pull/1

Various fixes required to get GISS-GC built with GEOS-Chem v14.3.1.

### CODE REVIEW

File: model/geos-chem/CMakeScripts/GC-ConfigureClassic.cmake
Comment (joewallwork) [inline model/geos-chem/CMakeScripts/GC-ConfigureClassic.cmake:1]: @ltmurray Is this the place where the code is purposefully broken in commit da5632768e8ef8cb4dd597c0a0a45070bb00207e?

File: config/compiler.intel.mk [hunk 1]
```diff
@@ -10,9 +10,9 @@ LFLAGS = -O2 -ftz
 CPPFLAGS += -DCOMPILER_Intel8 -DCONVERT_BIGENDIAN
 F90_VERSION = $(shell $(F90) -v 2>&1)
 ifeq ($(MP),YES)
-FFLAGS += -openmp
-F90FLAGS += -openmp
-LFLAGS += -openmp
+FFLAGS += -qopenmp
+F90FLAGS += -qopenmp
+LFLAGS += -qopenmp
```
Comment (joewallwork) [inline config/compiler.intel.mk:18]: This is required for more recent versions of the Intel compiler.

File: decks/Makefile [hunk 2]
```diff
@@ -127,29 +129,34 @@ ifeq ($(GC),YES)
...
-	cp $(MODEL_DIR)/geos-chem/build/src/GEOS-Chem/KPP/fullchem/libKPP.a $(MODEL_DIR)/geos-chem/lib/
-	cp $(MODEL_DIR)/geos-chem/build/src/GEOS-Chem/GeosCore/libGeosCore.a $(MODEL_DIR)/geos-chem/lib/
-	cp $(MODEL_DIR)/geos-chem/build/src/GEOS-Chem/GeosUtil/libGeosUtil.a $(MODEL_DIR)/geos-chem/lib/
-	cp $(MODEL_DIR)/geos-chem/build/src/GEOS-Chem/GeosUtil/libJulDay.a $(MODEL_DIR)/geos-chem/lib/
-	cp $(MODEL_DIR)/geos-chem/build/src/GEOS-Chem/NcdfUtil/libNcdfUtil.a $(MODEL_DIR)/geos-chem/lib/
-	cp $(MODEL_DIR)/geos-chem/build/src/Cloud-J/src/Core/libCloudJ_Core.a $(MODEL_DIR)/geos-chem/lib/
+	mkdir -p $(GC_BUILD_DIR)
+	mkdir -p $(GC_LIB_DIR)
+	cd $(GC_BUILD_DIR) && cmake .. -DINSTALLCOPY=.. \
+		-DOMP=$(MP) \
```
Comment (joewallwork) [inline decks/Makefile:135]: `MP` is GISS' flag determining whether to run with OpenMP (off by default). This line ensures that the same configuration is used within GEOS-Chem, too.

File: decks/Makefile [hunk 3]
```diff
@@ -127,29 +129,34 @@ ifeq ($(GC),YES)
...
+		-DCMAKE_Fortran_FLAGS="" \
+		-DCMAKE_EXE_LINKER_FLAGS="-nostartfiles -nofor-main"
```
Comment (joewallwork) [inline decks/Makefile]: This line was required to fix errors of the form
```
for_main.c:(.text+0x2e): undefined reference to `MAIN__'
```
The flag `-nofor-main` tells the Fortran compiler that the main program is not written in Fortran. Relatedly, the flag `-nostartfiles` prevents the compiler from using standard start-up files when linking.

File: decks/Makefile
Comment (joewallwork) [inline decks/Makefile:1]: Most of the changes in this file are to neaten the formatting. However, there are some key changes to get the build working - see comments.

File: config/compiler.gfortran.mk [hunk 4]
```diff
@@ -26,7 +26,7 @@ FMAKEDEP = $(SCRIPTS_DIR)/sfmakedepend
 CPPFLAGS += -DCOMPILER_G95
 FFLAGS = -g -cpp -fconvert=big-endian -O2 -fno-range-check -fallow-argument-mismatch
 F90FLAGS = -g -cpp -fconvert=big-endian -O2 -fno-range-check -ffree-line-length-none -fallow-argument-mismatch
-LFLAGS =
+LFLAGS = -nostartfiles -Wno-main
```
Comment (joewallwork) [inline config/compiler.gfortran.mk]: These (and others) need to be wrapped in `ifdef`s for only when we are compiling with GC.

**Comment by joewallwork** [comment]
Note that https://github.com/fetch4/GISS-GC/pull/3/commits/7de3cc9ed001b795a1c01773189a22b69f56bcaa means that this PR also addresses #1.

**Comment by joewallwork** [comment]
This PR is currently blocked by #8.

**Comment by joewallwork** [comment]
Continuous integration testing that GISS-GC can be successfully built is now implemented in #16, which branches off of the one here. If the reviewer is happy with the changes here then I think we can safely merge.

**Review by ltmurray — APPROVED** [review]
Approved

**Pull request merged**